components/movementsensor/wheeledodometry/wheeled_odometry.go

```diff
@@ -0,0 +1,299 @@
+	}
+
+	newBase, _ := base.FromDependencies(deps, newConf.Base)
+	props, err := newBase.Properties(ctx, nil)
+	o.baseWidth = props.WidthMeters
+	o.wheelCircumference = props.WheelCircumferenceMeters
+	o.orientation.Yaw = 0
+	o.math(context.Background())
+
+	return o, nil
+}
+
+func (o *odometry) AngularVelocity(ctx context.Context, extra map[string]interface{}) (spatialmath.AngularVelocity, error) {
+	o.mu.Lock()
+	defer o.mu.Unlock()
+	return o.angularVelocity, nil
+}
+
+func (o *odometry) LinearAcceleration(ctx context.Context, extra map[string]interface{}) (r3.Vector, error) {
+	o.mu.Lock()
+	defer o.mu.Unlock()
+	return o.acceleration, nil
+}
+
+func (o *odometry) Orientation(ctx context.Context, extra map[string]interface{}) (spatialmath.Orientation, error) {
+	o.mu.Lock()
+	defer o.mu.Unlock()
+	ov := &spatialmath.OrientationVector{Theta: o.orientation.Yaw, OX: 0, OY: 0, OZ: 1}
+	return ov, nil
+}
+
+func (o *odometry) CompassHeading(ctx context.Context, extra map[string]interface{}) (float64, error) {
+	o.mu.Lock()
+	defer o.mu.Unlock()
+	return o.orientation.Yaw, nil
+}
+
+func (o *odometry) LinearVelocity(ctx context.Context, extra map[string]interface{}) (r3.Vector, error) {
+	o.mu.Lock()
+	defer o.mu.Unlock()
+	return o.linearVelocity, nil
+}
+
+func (o *odometry) Position(ctx context.Context, extra map[string]interface{}) (*geo.Point, float64, error) {
+	o.mu.Lock()
+	defer o.mu.Unlock()
+	return geo.NewPoint(o.position.X, o.position.Y), o.position.Z, nil
+}
+
+func (o *odometry) Readings(ctx context.Context, extra map[string]interface{}) (map[string]interface{}, error) {
+	return movementsensor.Readings(ctx, o, extra)
+}
+
+func (o *odometry) Accuracy(ctx context.Context, extra map[string]interface{}) (map[string]float32, error) {
+	return map[string]float32{}, movementsensor.ErrMethodUnimplementedAccuracy
+}
+
+func (o *odometry) Properties(ctx context.Context, extra map[string]interface{}) (*movementsensor.Properties, error) {
+	return &movementsensor.Properties{
+		LinearVelocitySupported:     true,
+		AngularVelocitySupported:    true,
+		OrientationSupported:        true,
+		PositionSupported:           true,
+		LinearAccelerationSupported: false,
+	}, nil
+}
+
+func (o *odometry) Close(ctx context.Context) error {
+	o.cancelFunc()
+	o.activeBackgroundWorkers.Wait()
+	return nil
+}
+
+func (o *odometry) math(ctx context.Context) {
+	o.activeBackgroundWorkers.Add(1)
+	utils.PanicCapturingGo(func() {
+		defer utils.UncheckedErrorFunc(func() error {
+			o.mu.Lock()
+			defer o.mu.Unlock()
+			return errors.New("math failed")
+		})
+		defer o.activeBackgroundWorkers.Done()
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+			}
+			leftMove, _ := o.leftMotors[0].IsMoving(ctx)
+			rightMove, _ := o.rightMotors[0].IsMoving(ctx)
+			if leftMove || rightMove {
+				// sleep for the desired time interval
+				time.Sleep(time.Duration(o.timeInterval) * time.Millisecond)
```

Comment:
implement this as a ticker in your select tcase with the time interval

Reply:
looked into using a ticker and I'm not sure about it because I don't want it to return when the time interval is done, I want it to just not poll the motor position for that amount of time, but I want the loop to continue if that makes sense